cmd/serverless/main.go

```diff
@@ -297,7 +297,17 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 
 	// start the proxy if needed
 	// as this feature is experimental, this should be the only entrypoint (easier to remove)
-	_ = proxy.Start("127.0.0.1:9000", "127.0.0.1:9001", &invocationlifecycle.ProxyProcessor{})
+	if serverlessDaemon.MetricAgent != nil {
+		_ = proxy.Start(
+			"127.0.0.1:9000",
+			"127.0.0.1:9001",
+			&invocationlifecycle.ProxyProcessor{
+				ExtraTags:     serverlessDaemon.ExtraTags,
+				MetricChannel: serverlessDaemon.MetricAgent.GetMetricChannel(),
+			})
+	} else {
+		log.Error("The metric agent has not been set. Unable to start the proxy")
```

Comment:
Maybe we want to check for the env variable before throwing the error in the log?